### PR TITLE
Readthedocs integration

### PR DESCRIPTION
**Pull request by mchesnay**
### Linked issues

_none_

### Summarize your change.

Update the documentation to allow publication on [readthedocs](https://aswf-openrv.readthedocs.io/en/latest/).

### Describe the reason for the change.

Simplify access to the Open RV documentation through readthedocs. This should improve legibility and searchability.

### Describe what you have tested and on which operating system.

Couldn't test live on the read the docs system since it integrates directly with the github repo. 
Did use sphinx to build and render the docs locally. While the pipeline is obviously different, the rendering is similar. Local rendering was successful.

### Add a list of changes, and note any that might need special attention during the review.

Files required by readthedocs:
- .readthedocs.yml
- docs/conf.py
- docs/requirements.txt

Files modified to create the documentation structure on readthedocs, aka the TOC:
- docs/index.md

The deleted files were previously used to provide a sort of toc for navigating within github, but are no longer required for readthedocs.

### CODE REVIEW

**Review by geffrak — APPROVED** [review]

File: docs/requirements.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+myst-parser
```
Comment (geffrak) [inline docs/requirements.txt:1]: Can you annotate the requirement like it's done in this file?

https://github.com/AcademySoftwareFoundation/OpenRV/blob/main/requirements.txt

**Review by geffrak — CHANGES_REQUESTED** [review]

**Comment by JeanChristopheMorinPerso** [comment]
RTD can be configured to build previews for PRs: https://docs.readthedocs.io/en/latest/pull-requests.html.

**Comment by geffrak** [comment]
> RTD can be configured to build previews for PRs: https://docs.readthedocs.io/en/latest/pull-requests.html.

I don't think we have the permissions to do it. Would it be you?

**Review by geffrak — APPROVED** [review]

**Comment by JeanChristopheMorinPerso** [comment]
I don't have access to the project settings in RTD. But you could probably ask to be added as a maintainer over there.

**Review by compaga — CHANGES_REQUESTED** [review]
You should update the documentation link on the ReadMe.md at the root to point to ReadTheDocs.

**Review by compaga — APPROVED** [review]

**Pull request merged**